addons/sale_timesheet/migrations/11.0.1.0/post-migration.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2018 Tecnativa - Vicent Cubells
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openupgradelib import openupgrade
+from psycopg2.extensions import AsIs
+
+
+def migrate_sale_line_tasks(env):
+    openupgrade.logged_query(
+        env.cr, """
+            UPDATE sale_order_line sol
+            SET task_id = po.task_id
+            FROM procurement_order po
+            WHERE sol.id = po.sale_line_id
+        """)
+
+
+def migrate_project_sale_line(env):
+    openupgrade.logged_query(
+        env.cr, """
+            UPDATE project_project pj
+            SET sale_line_id = sol.id
+            FROM sale_order_line sol, product_product pd, sale_order so
+            WHERE sol.product_id = pd.id
+            AND sol.order_id = so.id
+            AND so.analytic_account_id = pj.id
+        """)
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    openupgrade.map_values(
+        env.cr, openupgrade.get_legacy_name('service_type'), 'service_type',
+        [('task', 'timesheet')], table='product_template',
+    )
+    # Update invoice_policy
+    env['product.template'].search([
+        ('type', '=', 'service'),
+    ])._compute_service_policy()
```

Comment:
I miss before doing this computation, to check if `invoice_policy` should change according old fields.

Reply:
I don't think so. `invoice_policy` is a field that not change.

Reply:
OK